### PR TITLE
[23.11] nixos/snapper: Add peristentTimer option to config

### PR DESCRIPTION
**Pull request by mrkline**
Defaults to false, but allows users to enable it for machines that aren't on persistently (e.g., laptops, home PCs).

Backport of 2c55e03 (#306909) but using `lib.mdDoc` to be consistent with the rest of the options on this branch.

### CODE REVIEW

**Comment by mrkline** [comment]
@tomberek if you've got the time to take a look over the backport as well, much appreciated!

**Comment by jian-lin** [comment]
Thanks for your contribution.

Here is some doc for manual backport: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#manually-backporting-changes.  Could you make some changes such as PR title and `git cherry-pick -x` according to that?

> using lib.mdDoc to be consistent with the rest of the options on this branch

This is indeed an disadvantage of [auto backport](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#automatically-backporting-changes), which I think is acceptable.  If you want an auto backport, I can help with that.

**Comment by mrkline** [comment]
AIUI `cherry-pick -x` adds a line referencing the commit it's coming from - I've done that (and referenced the PR).

And it's not a cherry-pick (neither manual nor auto-backported), as the commit message notes - I've changed it to match the rest of the module's args in 23.11

**Comment by jian-lin** [comment]
> AIUI cherry-pick -x adds a line referencing the commit it's coming from - I've done that (and referenced the PR).

I see you link that PR in the description.

What I ask is that you also reference the backported commit hash `ff0f4540c088913db38d9a5006b4a85e768c9ae5` in the commit message like what `git cherry-pick -x` gives you.

> as the commit message notes - I've changed it to match the rest of the module's args in 23.11

I see you add `lib.mkDoc` here.

I suggest an auto-backport because the addition of `lib.mkDoc` is, in my opinion, only cosmetic and the auto-backport references to both the PR and the commit hash properly.

> And it's not a cherry-pick (neither manual nor auto-backported)

Usually, a manual backport indicates modifying the `cherry-pick`ed commit, either in place or in a new commit.

> Could you make some changes such as PR title

Not sure if you missed this suggestion or you intend to not do this.  Anyway, I have changed it because I see no harm of it.  Hope you do not mind.

**Comment by mrkline** [comment]
> What I ask is that you also reference the backported commit hash

The PR description contains that as well; I assumed that it would be used as the commit message when the PR is merged.

> I suggest an auto-backport

I'm not a maintainer (yet!), so I don't think I have the powers to do so.

I force-pushed a commit generated from `cherry-pick -x`, hopefully that cleans things up. Apologies for any misunderstanding.

File: nixos/modules/services/misc/snapper.nix [hunk 1]
```diff
@@ -103,6 +103,18 @@ in
       '';
     };
 
+    persistentTimer = mkOption {
+      default = false;
+      type = types.bool;
+      example = true;
+      description = ''
+        Set the `persistentTimer` option for the
```
Comment (jian-lin) [inline nixos/modules/services/misc/snapper.nix:111]: nit: Should it be `Persistent`?  If so, this (and the one in the [borgbackup module](https://github.com/NixOS/nixpkgs/blob/00c95af548cbee2ec12a515ae409a65d651273f2/nixos/modules/services/backup/borgbackup.nix#L364)) can be fixed in a future PR.

**Review by jian-lin — APPROVED** [review]

**Comment by jian-lin** [comment]
> The PR description contains that as well; I assumed that it would be used as the commit message when the PR is merged.

It is not the case for two of three merge methods on GitHub.

> I'm not a maintainer (yet!), so I don't think I have the powers to do so.

I know and that's why I say I can help with that (adding the needed tag).  I should have stated that more clear.

Again, thanks.

File: nixos/modules/services/misc/snapper.nix (hunk 1 above)
Comment (mrkline) [reply nixos/modules/services/misc/snapper.nix:111]: Addressed with https://github.com/NixOS/nixpkgs/pull/314099

**Pull request merged**